src/components/input/Input.tsx

```diff
@@ -0,0 +1,113 @@
+import { PasswordOffIcon, PasswordOnIcon } from '@/libs/utils/Icon';
+import React, { useState } from 'react';
+import { FieldError, UseFormRegisterReturn } from 'react-hook-form';
+
+/**
+ * 재사용 가능한 입력 컴포넌트로, 텍스트, 비밀번호, 숫자 타입을 지원합니다.
+ * 비밀번호 필드에는 비밀번호 표시/숨기기 토글이 포함되어 있습니다.
+ *
+ * @example
+ * ```jsx
+ * const { register, handleSubmit, formState: { errors } } = useForm<FormData>({
+ *   resolver: yupResolver(schema),
+ * });
+ *
+ * const onSubmit: SubmitHandler<FormData> = (data) => {
+ *   console.log('폼 제출', data);
+ * };
+ *
+ * <Input
+ *   label="이메일"
+ *   name="email"
+ *   type="text"
+ *   placeholder="이메일을 입력해주세요"
+ *   register={register('email')}
+ *   error={errors.email}
+ *   maxWidth="640px"
+ * />
+ * ```
+ *
+ * @typedef {Object} InputProps
+ * @property {string} label - 입력 필드의 레이블
+ * @property {string} name - 입력 필드의 이름
+ * @property {'text' | 'password' | 'number'} [type='text'] - 입력 필드의 타입
+ * @property {string} placeholder - 입력 필드의 플레이스홀더 텍스트
+ * @property {UseFormRegisterReturn} register - react-hook-form의 register 함수 반환값
+ * @property {FieldError} [error] - 입력 필드의 에러 정보
+ * @property {string} [maxWidth='640px'] - (반응형을 위한)입력 필드 컨테이너의 최대 너비. 기본값 640px
+ * @author 김보미
+ */
+
+type InputProps = {
+  label: string;
+  name: string;
+  type?: 'text' | 'password' | 'number';
+  placeholder: string;
+  register: UseFormRegisterReturn;
+  error?: FieldError;
+  maxWidth?: string;
+};
```

Comment:
react-hook-form 고려해서 타입 지정 너무 잘해주셨네요!

저라면 내려주는 prop이 많기도 하고 확장성 측면에서 이렇게 했을 것 같아요!

```suggestion
type InputProps = InputHTMLAttributes<HTMLInputElement> & {
  label: string;
  type?: 'text' | 'password' | 'number';
  register: UseFormRegisterReturn;
  error?: FieldError;
  maxWidth?: string;
};

export default function Input({
  label,
  register,
  error,
  maxWidth = '640px',
  type = 'text',
  ...props
}: InputProps) {
  const { name, placeholder } = props;
```

막상 해보니까 두 개밖에 안 줄여지네요 ㅎㅎ 
보미님께서 더 나은 방법으로 선택해주시면 좋을 것 같습니다..!

Reply:
안그래도 프롭스 넘 많아서 걱정했는데 두 개라도 없애는게 낫죠 ㅎㅎ 반영하겠읍니다 !